File: 1.21.9/src/main/java/com/bleudev/nine_lifes/networking/payloads/ArmorStandHitEvent.java
package com.bleudev.nine_lifes.networking.payloads;

import com.bleudev.nine_lifes.networking.Packets;
import net.minecraft.network.RegistryByteBuf;
import net.minecraft.network.codec.PacketCodec;
import net.minecraft.network.packet.CustomPayload;
import net.minecraft.util.math.Vec3d;

public record ArmorStandHitEvent(Vec3d pos) implements CustomPayload {
    public static final CustomPayload.Id<ArmorStandHitEvent> ID = new CustomPayload.Id<>(Packets.ARMOR_STAND_HIT_EVENT);
    public static final PacketCodec<RegistryByteBuf, ArmorStandHitEvent> CODEC = PacketCodec.tuple(
            Vec3d.PACKET_CODEC, ArmorStandHitEvent::pos,
            ArmorStandHitEvent::new
    );

    @Override
    public Id<? extends CustomPayload> getId() {
        return ID;
    }
}

File: 1.21.10/src/main/kotlin/com/bleudev/nine_lifes/custom/packet/payload/ArmorStandHitEvent.kt
package com.bleudev.nine_lifes.custom.packet.payload

import com.bleudev.nine_lifes.custom.NineLifesPackets
import com.bleudev.nine_lifes.custom.packet.payload.interfaces.PacketPayloadCompanion
import net.minecraft.network.RegistryFriendlyByteBuf
import net.minecraft.network.codec.StreamCodec
import net.minecraft.network.protocol.common.custom.CustomPacketPayload
import net.minecraft.world.phys.Vec3

class ArmorStandHitEvent(val pos: Vec3): CustomPacketPayload {
    companion object : PacketPayloadCompanion<ArmorStandHitEvent> {
        override val idLocation = NineLifesPackets.ARMOR_STAND_HIT_EVENT
        override val codec: StreamCodec<RegistryFriendlyByteBuf, ArmorStandHitEvent> = StreamCodec.composite(
            Vec3.STREAM_CODEC, ArmorStandHitEvent::pos,
            ::ArmorStandHitEvent)
    }

    override fun type(): CustomPacketPayload.Type<out CustomPacketPayload> = id
}